gno.land/pkg/gnoland/node_inmemory.go

```diff
@@ -0,0 +1,147 @@
+package gnoland
+
+import (
+	"fmt"
+	"time"
+
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	tmcfg "github.com/gnolang/gno/tm2/pkg/bft/config"
+	"github.com/gnolang/gno/tm2/pkg/bft/node"
+	"github.com/gnolang/gno/tm2/pkg/bft/proxy"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/crypto/ed25519"
+	"github.com/gnolang/gno/tm2/pkg/db"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/gnolang/gno/tm2/pkg/p2p"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+type InMemoryNodeConfig struct {
+	PrivValidator         bft.PrivValidator // identity of the validator
+	Genesis               *bft.GenesisDoc
+	TMConfig              *tmcfg.Config
+	SkipFailingGenesisTxs bool
+	GenesisMaxVMCycles    int64
+}
+
+// NewMockedPrivValidator generate a new key
+func NewMockedPrivValidator() bft.PrivValidator {
```

Comment:
are you using this helper outside of this file? if no, remove this helper to keep the API smaller.

Reply:
it is actually used for testing in `pkg/integration/testing_node.go`